wurst/event/DamageModify.wurst

```diff
@@ -0,0 +1,327 @@
+/**
+	DamageModify
+	by Trokkin
+
+	API:
+
+	// Pretty much '(Damage d) -> ...'
+	interface ForDamageCallback
+		run(Damage d)
+
+	// these are called before the damage is dealt
+	// can be called multiple amount of times or not called at all
+	addDamageModifier(ForDamageCallback cb)
+	removeDamageModifier(ForDamageCallback cb)
+	
+	// these are called after the damage is dealt
```

Comment:
"These provide a callback API for after damage has occurred. Only one listener may be present per damage event."

Why the restriction on 1?

What's the use case for supporting both before and after? How much after does it occur - is it using a timer?

Reply:
Hm, do we have any DDS that shout its events more than once? I mentioned the restriction in contrast of damageModifiers, that can be called more than once.

it uses a timer in cases when incoming damage from event takes target's life beyond [.405; MAX_LIFE]. Otherwise, right before the actual damage. 

For example, imagine a spell that deals damage that doubles any effects on it's amount. And an item on it's caster that grants spell lifesteal.
Then any damageModifiers will modify that damage twice, but damageListeners - will be called only once, so you won't get twice as much life from that damage.
Same applies for 'pure' damage that isn't being modified at all, but works with lifesteal.

Reply:
For cases that my question warrants a useful explanation, I expect some changes in code comments